bash/scripts/commit_description_parser.sh

```diff
@@ -10,25 +10,47 @@ main() {
 }
 
 # parse-commit-description parses a commit description for any required sibling
-# to pass along to downstream job(s)...
+# commits to pass along to downstream job(s)...
 parse-commit-description() {
   description="${1}"
 
-  # Looks specifically for matches of '[rR]equires <repo>#<sha>',
-  # e.g., "requires builder#abc1234, Requires router#def5678"
-  reqs=$(echo "${description}" | grep -o "[Rr]equires [-a-z]*#[a-z0-9]*" | grep -o "[-a-z]*#[a-z0-9]*") || true
+  # Looks specifically for matches of '[rR]equires <repo>#<prNumber>',
+  # e.g., "requires builder#123, Requires router#567"
+  reqs=$(echo "${description}" | grep -o "[Rr]equires [-a-z]*#[0-9]*" | grep -o "[-a-z]*#[0-9]*") || true
 
-  # split on whitespace into array of '<repo>#<sha>' values
+  # split on whitespace into array of '<repo>#<prNumber>' values
   reqsArray=(${reqs// / })
 
-  # for each '<repo>#<sha>' value in array
+  # for each '<repo>#<prNumber>' value in array
   for i in "${reqsArray[@]}"; do
     # split on '#'
-    repoShaArray=(${i//#/ })
-    # echo '<uppercased repo>_SHA'=<sha> (with hyphens converted to underscores)
-    repoEnvVar=$(echo "${repoShaArray[0]//-/_}" | awk '{print toupper($0)}')_SHA
-    echo "${repoEnvVar}"="${repoShaArray[1]}"
+    repoPRNumberArray=(${i//#/ })
+    repoName="${repoPRNumberArray[0]}"
+    prNumber="${repoPRNumberArray[1]}"
+
+    # curl GH api for needed commit sha, providing name and issue number
+    sha=$(get-most-recent-pr-commit "${repoName}" "${prNumber}")
+
+    if [[ ${sha} =~ ^[a-f0-9]{5,40}$ ]]; then
+      # echo '<uppercased repo>_SHA'=<prNumber> (with hyphens converted to underscores)
+      repoEnvVar=$(echo "${repoName//-/_}" | awk '{print toupper($0)}')_SHA
+      echo "${repoEnvVar}"="${sha}"
+    else
+      echo "Commit sha for PR #${prNumber} in repo '${repoName}' not found."
+      return 1
+    fi
   done
 }
 
+get-most-recent-pr-commit() {
+  repoName="${1}"
+  prNumber="${2}"
+
+  curl \
+  -sSL \
+  --user deis-admin:"\${GITHUB_ACCESS_TOKEN}" \
+  "https://api.github.com/repos/deis/${repoName}/pulls/${prNumber}/commits" \
+  | docker run -i --rm kamermans/jq '.[].sha' | tail -n1
```

Comment:
I'm sure you anticipated this comment from me... it seems like this suite of scripts is now complex enough to warrant building a Go program to replace (at least part of) it. Thoughts on doing that eventually?